### PR TITLE
Add configury to enable spinlock.

### PR DESCRIPTION
**Pull request by bturrubiates**
@pmmccorm  @jsquyres 
Please review. 

Signed-off-by: Ben Turrubiates <bturrubiates@lanl.gov>

### CODE REVIEW

**Comment by bturrubiates** [comment]
See #798.

File: configure.ac [hunk 1]
```diff
@@ -91,6 +91,24 @@ dnl Checks for libraries
 AC_CHECK_LIB(pthread, pthread_mutex_init, [],
     AC_MSG_ERROR([pthread_mutex_init() not found.  libfabric requires libpthread.]))
 
+AC_ARG_ENABLE([spinlock],
+	[AC_HELP_STRING([--enable-spinlock],
+		[Enable spinlock support (default:disabled)])])
```
Comment (shefty) [inline configure.ac]: We should not ask the user to opt-in to using a higher performing synchronization mechanism.  If spinlock support is available, we should just use it.

Comment (bturrubiates) [inline configure.ac]: Okay. I'll fix sometime tomorrow.

**Comment by bturrubiates** [comment]
@shefty How's this?

**Pull request merged**